crates/red_knot_python_semantic/src/types.rs

```diff
@@ -824,6 +813,96 @@ impl<'db> From<&Type<'db>> for Type<'db> {
     }
 }
 
+/// Non-exhaustive enumeration of known classes (e.g. `builtins.int`, `typing.Any`, ...) to allow
+/// for easier syntax when interacting with very common classes.
+///
+/// Feel free to expand this enum if you ever find yourself using the same class in multiple
+/// places.
+/// Note: good candidates are any classes in `[crate::stdlib::CoreStdlibModule]`
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum KnownClass {
+    // Builtins
+    Bool,
+    Object,
+    Bytes,
+    Int,
+    Float,
+    Str,
+    List,
+    Tuple,
+    Set,
+    Dict,
+    // Typing
+    Any, // Not currently used, included to show support for non-builtins
+}
+
+impl<'db> KnownClass {
+    pub const fn as_str(&self) -> &'static str {
+        match self {
+            Self::Bool => "bool",
+            Self::Object => "object",
+            Self::Bytes => "bytes",
+            Self::Tuple => "tuple",
+            Self::Int => "int",
+            Self::Float => "float",
+            Self::Str => "str",
+            Self::Set => "set",
+            Self::Dict => "dict",
+            Self::List => "list",
+            Self::Any => "Any",
+        }
+    }
+
+    pub fn to_instance(&self, db: &'db dyn Db) -> Type<'db> {
+        self.to_class(db).to_instance(db)
+    }
+
+    pub fn to_class(&self, db: &'db dyn Db) -> Type<'db> {
+        match self {
+            Self::Any => types_symbol_ty(db, self.as_str()),
+            _ => builtins_symbol_ty(db, self.as_str()),
+        }
+    }
+
+    pub fn maybe_from_module(module: &Module, class_name: &str) -> Option<Self> {
+        let candidate = Self::from_name(class_name)?;
+        if candidate.check_module(module) {
+            Some(candidate)
+        } else {
+            None
+        }
+    }
+
+    pub fn from_name(name: &str) -> Option<Self> {
+        match name {
+            "bool" => Some(Self::Bool),
+            "object" => Some(Self::Object),
+            "bytes" => Some(Self::Bytes),
+            "tuple" => Some(Self::Tuple),
+            "int" => Some(Self::Int),
+            "float" => Some(Self::Float),
+            "str" => Some(Self::Str),
+            "set" => Some(Self::Set),
+            "dict" => Some(Self::Dict),
+            "list" => Some(Self::List),
+            "Any" => Some(Self::Any),
+            _ => None,
+        }
+    }
+
+    /// Private method checking if known class can be defined in the given module.
+    fn check_module(self, module: &Module) -> bool {
+        let is_builtin = module.search_path().is_standard_library() && module.name() == "builtins";
+        let is_typing = module.search_path().is_standard_library() && module.name() == "typing";
+        let is_typing_extensions =
+            module.search_path().is_standard_library() && module.name() == "typing_extensions";
+        match self {
+            Self::Any => is_typing || is_typing_extensions,
+            _ => is_builtin,
+        }
```

Comment:
this feels a bit repetitive, and it also feels like we're doing some work that isn't always necessary depending on what `self` is. Maybe something like this?

```suggestion
        let is_stdlib_module = module.search_path().is_standard_library();
        let module_name = module.name().as_str();
        match self {
            Self::Any => is_stdlib_module && matches!(module_name, "typing" | "typing_extensions")
            _ => is_stdlib_module && module_name == "builtins",
        }
```

I'd actually also prefer it if we tediously enumerated all the builtin classes rather than using `_` as a fallback branch. If we do it that way then the compiler will warn us when we add a new non-builtins branch but forget to update this method, which would lead to incorrect results.

Reply:
Actually we could also early return if we are not in stdlib as all cases in the close future would be in stdlib